crates/node-types/src/lib.rs

```diff
@@ -85,6 +100,40 @@ impl Default for BigBang {
     }
 }
 
+#[cfg(feature = "new-block")]
+impl BlockTx {
+    /// Create a new [`BlockTx`] to notify listeners of new blocks.
+    pub fn new() -> Self {
+        let (block_tx, _block_rx) = tokio::sync::watch::channel(());
+        Self(block_tx)
+    }
+
+    /// Notify listeners that a new block has been received.
+    ///
+    /// Note this is best effort and will still send even if there are currently no listeners.
+    pub fn notify(&self) {
+        let _ = self.0.send(());
+    }
+
+    /// Create a new [`BlockRx`] to listen for new blocks.
+    pub fn new_listener(&self) -> BlockRx {
+        BlockRx(self.0.subscribe())
+    }
+
+    /// Get the number of receivers listening for new blocks.
+    pub fn receiver_count(&self) -> usize {
+        self.0.receiver_count()
+    }
```

Comment:
This method is only used for testing right? I wonder if we should `#[cfg(test)]` this method to avoid having people downstream start relying on it, in case we wanna change the implementation of the `BlockTx` in the future?

Reply:
Went with not doing this per our discussion